### DISCUSSION TITLE
Add NBTCompound to ItemStack

### DISCUSSION THREAD

**Discussion by para-dise** [Minestom/Minestom#1432] [Help] [answered]
Hi,

Is there any way to add an NBTCompound  to an ItemStack?

I have the following code
```java
NBTCompound nbt = NBT.Compound(builder - > {
    for (StatType statType: this.getStats().getStats().keySet()) {
        if (this.getStats().getStats().get(statType) != 0) {
            builder.put(statType.toString(), NBT.Int(this.getStats().getStats().get(statType)));
        }
    }
});
````

I know perhaps I should've used Tags, but is there any way to add this NBTCompound to the ItemStack?

**Comment by TheMode** [comment, marked as answer]
One way would be to retrieve the item's meta NBT `item.meta().toNBT()`, modify it however you want, and then re-create an item using `ItemStack.fromNBT()`

**Reply by para-dise** [reply]
So there is no way to directly use the NBTCompound I created? I *need* to use the one generated from the builder?

**Reply by TheMode** [reply]
Well you can call `ItemStack.fromNBT()` directly with your NBT, but you would lose all the potential data already present in the item.

Otherwise you indeed need to merge the 2 compounds.
